app/src/main/java/com/hedvig/app/feature/embark/passages/previousinsurer/PreviousInsurerAdapter.kt

```diff
@@ -0,0 +1,73 @@
+package com.hedvig.app.feature.embark.passages.previousinsurer
+
+import android.graphics.drawable.PictureDrawable
+import android.net.Uri
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.RequestBuilder
+import com.hedvig.app.R
+import com.hedvig.app.databinding.ExpandableBottomSheetTitleBinding
+import com.hedvig.app.databinding.PreviousInsurerItemLayoutBinding
+import com.hedvig.app.util.GenericDiffUtilItemCallback
+import com.hedvig.app.util.extensions.inflate
+import com.hedvig.app.util.extensions.viewBinding
+
+class PreviousInsurerAdapter(
+    previousInsurers: List<PreviousInsurerData.PreviousInsurer>,
+    private val requestBuilder: RequestBuilder<PictureDrawable>,
+    private val onInsurerClicked: (String) -> Unit
+) : ListAdapter<PreviousInsurerItem, PreviousInsurerAdapter.PreviousInsurerViewHolder>(GenericDiffUtilItemCallback()) {
+
+    init {
+        submitList(listOf(PreviousInsurerItem.Header(text = "Insurers")) + previousInsurers.map { it.toListItem() })
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = when (viewType) {
+        R.layout.expandable_bottom_sheet_title -> PreviousInsurerViewHolder.Header(parent.inflate(R.layout.expandable_bottom_sheet_title))
+        R.layout.previous_insurer_item_layout -> PreviousInsurerViewHolder.InsurerViewHolder(parent.inflate(R.layout.previous_insurer_item_layout))
+        else -> throw Error("No view type found for: $viewType")
+    }
+
+    override fun onBindViewHolder(holder: PreviousInsurerViewHolder, position: Int) = when (holder) {
+        is PreviousInsurerViewHolder.InsurerViewHolder -> holder.bind(getItem(position) as PreviousInsurerItem.Insurer, requestBuilder, onInsurerClicked)
```

Comment:
Using the `abstract` method-variant allows this method to simply be defined as such:

```kotlin
holder.bind(getItem(position))
```